src/platform/plugins/shared/controls/public/controls/data_controls/options_list_control/get_options_list_control_factory.tsx

```diff
@@ -398,6 +398,34 @@ export const getOptionsListControlFactory = (): DataControlFactory<
         setSort: (sort: OptionsListSortingType | undefined) => {
           sort$.next(sort);
         },
+        selectAll: (keys: string[]) => {
+          const field = api.field$.getValue();
+          if (keys.length < 1 || !field) {
+            api.setBlockingError(
+              new Error(OptionsListStrings.control.getInvalidSelectionMessage())
+            );
+            return;
+          }
+
+          const selectedOptions = selectionsManager.api.selectedOptions$.getValue() ?? [];
+          const newSelections = keys.filter((value) => !selectedOptions.includes(value as string));
+          selectionsManager.api.setSelectedOptions([...selectedOptions, ...newSelections]);
```

Comment:
Why `filter` here? Why not just:

```suggestion
          const selectedOptions = selectionsManager.api.selectedOptions$.getValue() ?? [];
          selectionsManager.api.setSelectedOptions([...selectedOptions, ...keys]);
```

It shouldn't matter if `keys` includes the same values as `selectedOptions`, right? 🤔

Reply:
You're right. I can remove the filter. Addressed in [b3cec9a](https://github.com/elastic/kibana/pull/221010/commits/b3cec9a4fdf3abe5c0d33fbe8b68435f2d2fd64d)

Reply:
<img width="1127" alt="Screenshot 2025-05-29 at 11 16 41 AM" src="https://github.com/user-attachments/assets/8e7e390b-ccc0-41ec-8f72-ea103c39d517" />
Actually without the filter, it looks like it's duping the keys in the selection array.

Reply:
😆 Oops! I was thinking about object keys - my bad. If the selections were a set, we could spread without filtering - but you're right, since they are an array, this will duplicate stuff. Sorry 🤦